### ISSUE TITLE
Error moving to namespace . 

### ISSUE THREAD

**Issue by adrinux** [b-m-f/wg-pod#3]
wg-pod looks really handy but I'm getting an error when the move to the container namespace occurs. It seems that in my case wg-pod is trying to use . instead of an actual namespace and ip isn't playing.

A snippet of the journal:
```
Adding container wggenweb into WireGuard network defined in /etc/wireguard/moadmin.conf
Feb 18 17:22:52 momod wg-pod[13131]: Create temporary private key file for WireGuard interface at /run/containers/network/wggenweb.pkey
Feb 18 17:22:52 momod wg-pod[13131]: Added new WireGuard interface podman-ec19a2f
Feb 18 17:22:52 momod wg-pod[13131]: Error: problem when trying to move WireGuard interface podman-ec19a2f to namespace .
Feb 18 17:22:52 momod wg-pod[13131]:  RTNETLINK answers: Invalid argument
```

When starting this pod infra container on Ubuntu 21.10 via this systemd unit file:
```

[Unit]
Description=Podman wggenweb-pod.service
Documentation=man:podman-generate-systemd(1)
Wants=network.target
After=network-online.target
RequiresMountsFor= 
Requires=wggenweb-container.service
Before=wggenweb-container.service

[Service]
Environment=PODMAN_SYSTEMD_UNIT=%n
Restart=always
RestartSec=30
TimeoutStopSec=70
ExecStartPre=/bin/rm -f %t/%n.pid %t/%n.pod-id
ExecStartPre=/usr/bin/podman pod create \
--infra-conmon-pidfile %t/%n.pid \
--pod-id-file %t/%n.pod-id \
--publish 8080:8080 \
--name wggenweb \
--replace
ExecStart=/usr/bin/podman pod start --pod-id-file %t/%n.pod-id
ExecStartPost=/usr/local/bin/wg-pod join wggenweb \
/etc/wireguard/moadmin.conf \
--port-remapping 8080:8080
ExecStop=/usr/bin/podman pod stop --ignore --pod-id-file %t/%n.pod-id -t 10
ExecStopPost=/usr/bin/podman pod rm --ignore -f --pod-id-file %t/%n.pod-id
PIDFile=%t/%n.pid
Type=forking
KillMode=control-group

[Install]
WantedBy=multi-user.target default.target
```

Any suggestions on debugging would be welcome.

Edit: I suppose the obvious thing is to remove the ExecStartPost line and see what happens when I manually try to get the namespace etc.

**Comment by b-m-f** [comment]
Hi @adrinux ,

The namespace is detected in https://github.com/b-m-f/wg-pod/blob/bafdf737c8d5297cea92cb31fc9b4e63d944e910/pkg/podman/podman.go#L41 .

Can you try running the shell command manually?
The name variable is the name of the container.

**Comment by adrinux** [comment]
That gave an error "can't evaluate field NetworkSettings in type interface", but podman inspect name showed no NetworkSettings section so that makes sense.

I was passing the name of a pod infra container, looks like I should be passing the name of the container which has the app even though it's the infra container that publishes exposed ports.

I think I need to read a bit more about networking in pods...

I'll break things down and try a fresh run passing the app container.

**Comment by b-m-f** [comment]
I have not yet used the feature of grouping into a pod, so I have no tips.

Good luck though. If you find sth to make it work maybe we can add it to the Readme :)

**Comment by adrinux** [comment]
I reconsidered why I needed to be running these containers in a Pod and decided it wasn't necessary, so I've worked around this issue for now - though I'll likely return to it later.

I'll close this issue since wg-pod seems to be working as intended if I pass an app container to instead of the pod infra container.

(Still can't access my services over wireguard but I don't think that's wg-pod's fault.)

**Comment by b-m-f** [comment]
Maybe try setting the PersistentKeepalive setting.

Good luck

**Comment by adrinux** [comment]
> PersistentKeepalive setting

Nothing so simple - already had that enabled.

I went back to basics in a centos stream 9 vm on my workstation. Manually running much of it. I can access the grafana container I'm testing at the vm's IP or via wireguard (defaults to listening on all interfaces 0.0.0.0:3000).

But when to use wg-pod I keep hitting in an error at setting the default route:
```
Error: problem when setting the default route in cni-7819811d-2d52-07d2-6923-1bb3524e8f41 to go through podman-c71b3f7
RTNETLINK answers: File exists
```
Same error regardless of whether I pass -d to delete the default route.
ip route at this point:
```
default via 192.168.122.1 dev en0 proto static metric 100 
10.89.0.0/24 dev cni-podman1 proto kernel scope link src 10.89.0.1 
172.20.0.0/24 dev wg0 proto kernel scope link src 172.20.0.2 
192.168.122.0/24 dev en0 proto kernel scope link src 192.168.122.29 metric 100 
```

I did have one successful run of wg-pod after fat-fingering the wrong IP when setting up the wireguard interface on the server 172.20.2.0 vs 172.20.0.2 with the at client wg interface  172.20.0.5 - but no successful connection through wireguard after that.

I'm at a bit of a loss. Feel like my lack of networking knowledge is the root problem but nothing I read about setting this up seems to differ from what I'm doing.
